sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -1120,10 +1120,10 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
       messageParameters = Map("cost" -> cost))
   }
 
-  def unsupportedDataTypeError(dt: String): SparkUnsupportedOperationException = {
+  def unsupportedDataTypeError(typeName: String): SparkUnsupportedOperationException = {
```

Comment:
Could you pass a `DataType` to the function instead of a string, and avoid in-place conversions using;
- toString
- catalogString

Reply:
Will keep in mind.
Thanks, updated!!